xmake/modules/package/tools/cmake.lua

```diff
@@ -755,6 +723,92 @@ function _get_configs_for_install(package, configs, opt)
     end
 end
 
+function _get_default_flags(package, configs, buildtype, opt)
+    local cmake_default_flags = _g.cmake_default_flags and _g.cmake_default_flags[buildtype]
+    if not cmake_default_flags then
+        local tmpdir = path.join(os.tmpdir(), package:name(), package:mode())
+        local dummy_cmakelist = path.join(tmpdir, "CMakeLists.txt")
+
+        io.writefile(dummy_cmakelist, format([[
+    message("CMAKE_C_FLAGS is ${CMAKE_C_FLAGS}")
+    message("CMAKE_C_FLAGS_%s is ${CMAKE_C_FLAGS_%s}")
+
+    message("CMAKE_CXX_FLAGS is ${CMAKE_CXX_FLAGS}")
+    message("CMAKE_CXX_FLAGS_%s is ${CMAKE_CXX_FLAGS_%s}")
+
+    message("CMAKE_EXE_LINKER_FLAGS is ${CMAKE_EXE_LINKER_FLAGS}")
+    message("CMAKE_EXE_LINKER_FLAGS_%s is ${CMAKE_EXE_LINKER_FLAGS_%s}")
+
+    message("CMAKE_SHARED_LINKER_FLAGS is ${CMAKE_SHARED_LINKER_FLAGS}")
+    message("CMAKE_SHARED_LINKER_FLAGS_%s is ${CMAKE_SHARED_LINKER_FLAGS_%s}")
+
+    message("CMAKE_STATIC_LINKER_FLAGS is ${CMAKE_STATIC_LINKER_FLAGS}")
+    message("CMAKE_STATIC_LINKER_FLAGS_%s is ${CMAKE_STATIC_LINKER_FLAGS_%s}")
+        ]], buildtype, buildtype, buildtype, buildtype, buildtype, buildtype, buildtype, buildtype, buildtype, buildtype))
+
+        local runenvs = opt.envs or buildenvs(package)
+        local cmake = find_tool("cmake", {envs = runenvs})
+        local _configs = table.join(configs, "-S " .. path.directory(dummy_cmakelist), "-B " .. tmpdir)
+        local _, outdata = os.iorunv(cmake.program, _configs, {envs = runenvs})
```

Comment:
`local outdata, errdata = os.iorunv`, this outdata is errdata here.

Reply:
and we should add try/catch, `local outdata = try {function () return os.iorunv end}`

and outdata maybe is nil.

```lua
if outdata then
end
```

Reply:
> `local outdata, errdata = os.iorunv`, this outdata is errdata here.

i don't know why, but cmake message output are on errdata

Reply:
Probably because CMake fails to build the project as there's no project

Reply:
okay, i fixed it, message() by defaut output on stderr, i added STATUS and now it output on stdout